openghg/store/_boundary_conditions.py

```diff
@@ -160,66 +184,39 @@ def read_file(
         fn_current_parameters = locals().copy()  # Make a copy of parameters passed to function
         fn_input_parameters = {key: fn_current_parameters[key] for key in fn_input_parameters}
 
-        with open_dataset(filepath).chunk(chunks) as bc_data:
-            # Some attributes are numpy types we can't serialise to JSON so convert them
-            # to their native types here
-            attrs = {}
-            for key, value in bc_data.attrs.items():
-                try:
-                    attrs[key] = value.item()
-                except AttributeError:
-                    attrs[key] = value
-
-            author_name = "OpenGHG Cloud"
-            bc_data.attrs["author"] = author_name
-
-            metadata = {}
-            metadata.update(attrs)
-
-            metadata["species"] = species
-            metadata["domain"] = domain
-            metadata["bc_input"] = bc_input
-            metadata["author"] = author_name
-            metadata["processed"] = str(timestamp_now())
-
-            # Check if time has 0-dimensions and, if so, expand this so time is 1D
-            if "time" in bc_data.coords:
-                bc_data = update_zero_dim(bc_data, dim="time")
+        # Call appropriate standardisation function with input parameters
+        boundary_condition_data = parser_fn(**parser_input_parameters)
+        results: resultsType = defaultdict(dict)
 
+        for key, value in boundary_condition_data.items():
             # Currently ACRG boundary conditions are split by month or year
+            bc_data = value["data"]
             bc_time = bc_data["time"]
 
             start_date, end_date, period_str = infer_date_range(
                 bc_time, filepath=filepath, period=period, continuous=continuous
             )
 
             # Checking against expected format for boundary conditions
-            BoundaryConditions.validate_data(bc_data)
+            BoundaryConditions.validate_data(value["data"])
             data_type = "boundary_conditions"
 
-            metadata["start_date"] = str(start_date)
-            metadata["end_date"] = str(end_date)
-            metadata["data_type"] = data_type
-
-            metadata["max_longitude"] = round(float(bc_data["lon"].max()), 5)
-            metadata["min_longitude"] = round(float(bc_data["lon"].min()), 5)
-            metadata["max_latitude"] = round(float(bc_data["lat"].max()), 5)
-            metadata["min_latitude"] = round(float(bc_data["lat"].min()), 5)
-            metadata["min_height"] = round(float(bc_data["height"].min()), 5)
-            metadata["max_height"] = round(float(bc_data["height"].max()), 5)
+            additional_metadata["start_date"] = str(start_date)
```

Comment:
I think we would probably want these lines [204-217] which are adding the additional metadata details within the parse function. This may end up being something we put into a new utility function at some point so multiple parse functions can use it but I think in general we were trying to remove these kinds of direct metadata updates from with the data type classes.